axelor-human-resource/src/main/resources/views/Project.xml

```diff
@@ -9,11 +9,12 @@
     <extend target="//panel-dashlet[@name='tasksPerSprintPanel']">
       <insert position="before">
         <field name="sprint" domain="self.project.id = :id"
-          onChange="action-project-attrs-refresh-allocation-lines-on-sprint-change"/>
-        <field name="allocationPeriodSet" onSelect="action-project-attrs-set-periods-domain"
-          widget="tag-select"
+          onChange="action-project-attrs-refresh-allocation-lines-on-sprint-change" x-dirty="false"/>
```

Comment:
At onchange, remove from the periodList the periods not in the sprint before the refresh

And in the refresh manage the fact that sprint can be null.
If sprint and periods are null, filter on periods with fromDate > currentDate